api/src/main/java/io/opentelemetry/trace/propagation/B3Propagator.java

```diff
@@ -0,0 +1,201 @@
+      if (value == null || value.isEmpty()) {
+        logger.info(
+            "Missing or empty combined header: "
+                + COMBINED_HEADER
+                + ". Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      // must have between 2 and 4 hyphen delimieted parts:
+      //   traceId-spanId-sampled-parentSpanId (last two are optional)
+      // NOTE: we do not use parentSpanId
+      String[] parts = value.split(COMBINED_HEADER_DELIMITER);
+      if (parts.length < 2 || parts.length > 4) {
+        logger.info(
+            "Invalid combined header '" + COMBINED_HEADER + ". Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      traceId = parts[0];
+      if (!isTraceIdValid(traceId)) {
+        logger.info(
+            "Invalid TraceId in B3 header: "
+                + COMBINED_HEADER
+                + ". Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      spanId = parts[1];
+      if (!isSpanIdValid(spanId)) {
+        logger.info(
+            "Invalid SpanId in B3 header: "
+                + COMBINED_HEADER
+                + ". Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      sampled = parts.length == 3 ? parts[2] : null;
+    } else {
+      traceId = getter.get(carrier, TRACE_ID_HEADER);
+      if (!isTraceIdValid(traceId)) {
+        logger.info(
+            "Invalid TraceId in B3 header: "
+                + TRACE_ID_HEADER
+                + "'. Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      spanId = getter.get(carrier, SPAN_ID_HEADER);
+      if (!isSpanIdValid(spanId)) {
+        logger.info(
+            "Invalid SpanId in B3 header: "
+                + SPAN_ID_HEADER
+                + "'. Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      sampled = getter.get(carrier, SAMPLED_HEADER);
+    }
+
+    return buildSpanContext(traceId, spanId, sampled);
+  }
+
+  private static SpanContext buildSpanContext(String traceId, String spanId, String sampled) {
+    try {
+      TraceFlags traceFlags =
+          TraceFlags.builder()
+              .setIsSampled(
+                  TRUE_INT.equals(sampled)
+                      || Boolean.parseBoolean(sampled)) // accept either "1" or "true"
+              .build();
+
+      return SpanContext.createFromRemoteParent(
+          TraceId.fromLowerBase16(traceId, 0),
+          SpanId.fromLowerBase16(spanId, 0),
+          traceFlags,
+          TraceState.getDefault());
+    } catch (Exception e) {
+      logger.log(Level.INFO, "Error parsing B3 header. Returning INVALID span context.", e);
+      return SpanContext.getInvalid();
+    }
+  }
+
+  private static boolean isTraceIdValid(String value) {
+    return !(value == null || value.isEmpty() || value.length() > MAX_TRACE_ID_LENGTH);
```

Comment:
The first two checks could leverage the `StringUtils.isNullOrEmpty(value)` method

Reply:
Thanks - I've pushed an update to use `StringUtils`.